chain/events/filter/event.go

```diff
@@ -344,16 +346,41 @@ func (m *EventFilterManager) Revert(ctx context.Context, from, to *types.TipSet)
 		load:  m.loadExecutedMessages,
 	}
 
-	// TODO: could run this loop in parallel with errgroup if there are many filters
+	tsAddressResolver := m.createTipSetCachedAddressResolver()
+
+	g, ctx := errgroup.WithContext(ctx)
 	for _, f := range m.filters {
-		if err := f.CollectEvents(ctx, tse, true, m.AddressResolver); err != nil {
-			return err
-		}
+		f := f
+		g.Go(func() error {
+			return f.CollectEvents(ctx, tse, true, tsAddressResolver)
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		return err
 	}
 
 	return nil
 }
 
+func (m *EventFilterManager) createTipSetCachedAddressResolver() func(ctx context.Context, emitter abi.ActorID, ts *types.TipSet) address.Address {
```

Comment:
Subjective: 
* This could be a function (in FP sense): takes an address resolver and returns a cached version of it.

Then:
* One could wrap the struct field resolver with it at EventFilterManager.
* This would avoid multiple calls to `createTipSetCachedAddressResolver` in favour of singular cached address resolver for the lifetime of event filter manager (assuming this is desirable for performance reasons, though watch out for unbounded map size growth).

Reply:
Well, I'm only caching `abi.ActorID => address.Address`, not including the tipset, so I only want it cached for the particular tipset and then discarded since the mapping may vary between tipsets. We could potentially add tipset to the cache, make it an LRU to conserve some memory, but I feel like we have too many permanent caches across the codebase already that contribute to memory bloat.

Reply:
I've turned it from a method to a function that takes a resolver and added some docs to it to clarify its intended usage.